src/utils.js

```diff
@@ -238,3 +237,58 @@ export function parseCspPolicy(policy) {
 
   return parsedPolicy;
 }
+
+/**
+ * Determines if the source text is minified.
+ * Using the percentage no. of the indented lines from a sample set of lines
+ * to determine if the js file is minified.
+ * Inspired by code for the Firefox Developer Toolbar.
+ */
+export function couldBeMinifiedCode(code) {
+  // Number of lines to look at, taken from the head of the code.
+  const sampleSize = 30;
+
+  // Threshold in percent of indented lines to mark a file as not
+  // minified.
+  const indentCountThreshold = 20; // percentage
+
+  // Lenght of a line that looks suspicious of being minified
+  const hugeLinesLength = 500;
+
+  // Number of huge lines to also mark a file as potentially minified
+  // Hint: Minified AngularJS has 12 lines, jQuery 4
+  const hugeLinesThreshold = 4;
+
+  let lineEndIndex = 0;
+  let lineStartIndex = 0;
+  let lines = 1;
+  let indentCount = 0;
+  let hugeLinesCount = 0;
+
+  // Strip comments.
+  const normalizedCode = code.replace(/\/\*[\S\s]*?\*\/|\/\/.+/g, '');
+
+  while (lines < sampleSize) {
+    lineEndIndex = normalizedCode.indexOf('\n', lineStartIndex);
```

Comment:
What if a file has no `\n`?

Reply:
Then it's still matched and could very well be a minified code. Do you mean, if there are absolutely no `\n` we should watch out for longer lines? Or are you referring to hacks like https://github.com/mozilla/addons-server/blob/master/static/js/lib/semicolon.js to fix some compiled code?

Reply:
You're going to break out of the loop if no `\n` is found in the entire file. Whilst the check will return true currently you probably want a test to cover this eventually in-case the return logic changes in some way.

Reply:
I'll add a test

Reply:
Fixed in e05bb91de236bab06eb62021fb4ed4ea9e700853